apps/src/applab/designElements/textarea.jsx

```diff
@@ -231,7 +231,7 @@ export default {
   onPropertyChange: function (element, name, value) {
     switch (name) {
       case 'value':
-        element.innerHTML = value;
+        element.innerHTML = utils.escapeHtml(value);
```

Comment:
I see that in `textInput.jsx` that `element.value` is just assigned `value`. I wonder if we can just bypass the use of `innerHtml` and simplify the code.

Reply:
Then at [this line](https://github.com/code-dot-org/code-dot-org/blob/4b07aafa6e98449535f7fe86c8b11d3e8e35f08a/apps/src/applab/designElements/textarea.jsx#L50) you could assign the `initialValue` to `element.value`.

Reply:
hmm good q...I think the difference is that a "Text Area" in applab is actually just a `<div>`, which doesn't have a HTML attribute called value, so our approach is different. Text inputs are `<input>` elements, which have an HTML attribute called value (ie, the current value of what is in the input). There may be alternative/cleaner approaches to what we're doing currently, but I don't think we can just standardize them all to use `value`?

Reply:
> hmm good q...I think the difference is that a "Text Area" in applab is actually just a `<div>`, which doesn't have a HTML attribute called value, so our approach is different. Text inputs are `<input>` elements, which have an HTML attribute called value (ie, the current value of what is in the input). There may be alternative/cleaner approaches to what we're doing currently, but I don't think we can just standardize them all to use `value`?

That makes sense - thanks! I wonder then if we can just remove the `value` property from 'textArea', kinda undoing what was done in 'textArea' in this [past PR](https://github.com/code-dot-org/code-dot-org/pull/21389). I asked curriculum about this in the Slack thread you mentioned above.

Reply:
It's hard to tell for sure if we ever would have advertised this in our docs b/c I think our version history only goes back a couple of years

Reply:
Documenting here our discussion offline - we saw that if we [removed the 'textArea' `value` property](https://github.com/code-dot-org/code-dot-org/pull/53046), there is a regression when the user is in Design mode seen in an [integration test in `ec_design.js`](https://github.com/code-dot-org/code-dot-org/blob/6580b67d68b72af16c61a6179b3f6896cc9ca332/apps/test/integration/levelSolutions/applab1/ec_design.js#L709-L716) - they are no longer able to enter newlines in a text area in Design mode. Note that in code mode, users are able to run `setProperty('textArea1', 'text', 'text1\ntext2')` but in Design mode, the <br> is not able to be inserted. With this, I think your solution makes a lot of sense - thanks for taking time to look into this with me!